src/spyglass/common/common_dandi.py

```diff
@@ -0,0 +1,175 @@
+import os
+from uuid import uuid4
+
+import dandi.organize
+import dandi.validate
+import datajoint as dj
+import fsspec
+import h5py
+import pynwb
+from dandi.consts import known_instances
+from dandi.dandiapi import DandiAPIClient
+from dandi.validate_types import Severity
+from fsspec.implementations.cached import CachingFileSystem
+
+from spyglass.common.common_lab import LabMember
+from spyglass.common.common_usage import Export
+from spyglass.settings import export_dir
+from spyglass.utils.dj_helper_fn import _resolve_external_table
+from spyglass.utils.dj_mixin import SpyglassMixin
+
+dev_instance = known_instances["dandi-staging"]
+
+schema = dj.schema("common_dandi")
+
+
+@schema
+class DandiPath(SpyglassMixin, dj.Manual):
+    definition = """
+    -> Export.File
+    ---
+    dandiset_id: str
+    filename: varchar(255)
+    dandi_path: varchar(255)
+    """
+
+    def fetch_file_from_dandi(self, key: dict):
+        key = (self & key).fetch1("KEY")
+        dandiset_id = str((self & key).fetch1("dandiset_id"))
+        dandi_path = (self & key).fetch1("dandi_path")
+
+        # get the s3 url from Dandi
+        with DandiAPIClient(
+            dandi_instance=dev_instance,
+        ) as client:  # TODO: this is the dev server of dandi
+            asset = client.get_dandiset(dandiset_id).get_asset_by_path(
+                dandi_path
+            )
+            s3_url = asset.get_content_url(follow_redirects=1, strip_query=True)
+
+        # stream the file from s3
+        # first, create a virtual filesystem based on the http protocol
+        fs = fsspec.filesystem("http")
+
+        # create a cache to save downloaded data to disk (optional)
+        fs = CachingFileSystem(
+            fs=fs,
+            cache_storage=f"{export_dir}/nwb-cache",  # Local folder for the cache
```

Comment:
I don't know if each paper needs it's own cache, since this function would typically be used in a filesystem where the export wasn't created.  

My thought was a single cache directory for reading dandi-uploaded files